boards/trellis_m4/src/lib.rs

```diff
@@ -29,6 +31,11 @@ use keypad::{keypad_new, keypad_struct};
 /// Number of Neopixels on the device
 pub const NEOPIXEL_COUNT: usize = 32;
 
+/// Define the `DmacRegisters` type which impl's the `hal::dma::controller::Registers`
+/// trait and can be used with a DMA driver.
+#[cfg(feature = "unproven")]
+decl_dmac_registers!(DmacRegisters, dmac::DMAC, dmac::RegisterBlock);
```

Comment:
This macro is defining a newtype, `DmacRegisters`, that wraps the `svd2rust`-generated DMAC registers, and then impls the `hal::dma::controller::Registers` trait.

It can't impl the trait directly on `DMAC`, because `DMAC` is defined in the PAC and the trait is defined in the HAL.